File: src/main/java/com/parrit/transformers/PairingHistoryTransformer.java
package com.parrit.transformers;

import com.parrit.DTOs.PairingHistoryDTO;
import com.parrit.DTOs.PersonDTO;
import com.parrit.entities.PairingHistory;

import java.text.SimpleDateFormat;
import java.util.*;
import java.util.stream.Collectors;

public class PairingHistoryTransformer {

    private static SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");

    static {
        simpleDateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
    }

    public static PairingHistoryDTO transform(PairingHistory pairingHistory) {
        String pairingTime = simpleDateFormat.format(pairingHistory.getTimestamp());
        List<PersonDTO> personDTOs = PersonTransformer.transform(pairingHistory.getPeople());
        PairingHistoryDTO pairingHistoryDTO = new PairingHistoryDTO(pairingTime, personDTOs, pairingHistory.getPairingBoardName());
        return pairingHistoryDTO;
    }

    public static List<PairingHistoryDTO> transform(List<PairingHistory> pairingHistories) {
        if(pairingHistories == null || pairingHistories.isEmpty()) return Collections.emptyList();
        return pairingHistories.stream()
            .map(PairingHistoryTransformer::transform)
            .collect(Collectors.toList());
    }

}

File: src/main/kotlin/com/parrit/transformers/PairingHistoryTransformer.kt
package com.parrit.transformers

import com.parrit.DTOs.PairingHistoryDTO
import com.parrit.entities.PairingHistory

import java.text.SimpleDateFormat
import java.util.*
import java.util.stream.Collectors

object PairingHistoryTransformer {

    private val simpleDateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ")

    init {
        simpleDateFormat.timeZone = TimeZone.getTimeZone("UTC")
    }

    fun transform(pairingHistory: PairingHistory): PairingHistoryDTO {
        val pairingHistoryDTO = PairingHistoryDTO()
        pairingHistoryDTO.pairingTime = simpleDateFormat.format(pairingHistory.timestamp)
        pairingHistoryDTO.people = PersonTransformer.transform(pairingHistory.people)
        pairingHistoryDTO.pairingBoardName = pairingHistory.pairingBoardName
        return pairingHistoryDTO
    }

    fun transform(pairingHistories: List<PairingHistory>?): List<PairingHistoryDTO> {
        if (pairingHistories == null || pairingHistories.isEmpty()) return emptyList()
        return pairingHistories.map { transform(it) }
    }

}
